server/src/auth_service.rs

```diff
@@ -606,12 +613,20 @@ pub(crate) fn check_if_token_is_valid<Backend: BackendHandler>(
             token.header().algorithm
         )));
     }
-    let jwt_hash = default_hash(token_str);
-    if state.jwt_blacklist.read().unwrap().contains(&jwt_hash) {
-        return Err(ErrorUnauthorized("JWT was logged out"));
+
+    // Check if user is enabled
+    let user_id = UserId::new(&token.claims().user);
+    if !state
+        .get_tcp_handler()
+        .is_user_login_enabled(&user_id)
+        .await
+        .map_err(|_| ErrorUnauthorized("Could not verify user status"))?
```

Comment:
Propagate the error message here, so we know _why_ we couldn't verify the user status

Reply:
And maybe have a match for EntityNotFound with a custom error message.